### PR TITLE
[release/2.8] Add code to handle pagination of parts. Fixes max layer size of 10GB bug

### PR DESCRIPTION
**Pull request by DavidSpek**
- backport of https://github.com/distribution/distribution/pull/2815

### CODE REVIEW

**Comment by wy65701436** [comment]
@DavidSpek can you please make the CI pass?

**Comment by flavianmissi** [comment]
@DavidSpek this should fix the CI: https://github.com/distribution/distribution/pull/3843/files

**Comment by DavidSpek** [comment]
Thanks for the reviews. I'm traveling at the moment but I'll try and fix this PR later today or tomorrow.

**Comment by DavidSpek** [comment]
@flavianmissi Should be good for reviewing now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       run: |
           sudo apt-get -q update
           sudo -E apt-get -yq --no-install-suggests --no-install-recommends install python2-minimal
-          cd /tmp && go get -u github.com/vbatts/git-validation
+          cd /tmp && go install github.com/vbatts/git-validation@latest
```
Comment (thaJeztah) [inline .github/workflows/ci.yml:34]: FWIW; this one was already included in https://github.com/distribution/distribution/commit/2392893bcfe7de3010c66626db46f59ab520c9a0, so should dissolve when rebasing.

**Review by milosgajdos — APPROVED** [review]

**Comment by milosgajdos** [comment]
PTAL @thaJeztah @corhere @squizzi

**Pull request merged**